### PR TITLE
Fix When list_datasets fails, the UI shows error message but also shows "No data matches..." - it should only show the error

### PR DESCRIPTION
**Pull request by Taras-Hlukhovetskyi**
https://iguazio.atlassian.net/browse/ML-7162

### CODE REVIEW

File: src/components/JobWizard/JobWizardSteps/JobWizardFunctionSelection/JobWizardFunctionSelection.js [hunk 1]
```diff
@@ -280,7 +289,13 @@ const JobWizardFunctionSelection = ({
       activeTab === FUNCTIONS_SELECTION_HUB_TAB &&
       (isEmpty(hubFunctions) || isEmpty(hubFunctionsCatalog))
     ) {
-      dispatch(functionsActions.fetchHubFunctions()).then(templatesObject => {
+      dispatch(
+        functionsActions.fetchHubFunctions(
+          null,
+          { ui: { setRequestErrorMessage: setHubFunctionsRequestErrorMessage } }
+        )
+      ).then(templatesObject => {
```
Comment (illia-prokopchuk) [inline src/components/JobWizard/JobWizardSteps/JobWizardFunctionSelection/JobWizardFunctionSelection.js:297]: remove comment

File: src/components/JobWizard/JobWizardSteps/JobWizardFunctionSelection/JobWizardFunctionSelection.js [hunk 2]
```diff
@@ -280,7 +289,13 @@ const JobWizardFunctionSelection = ({
       activeTab === FUNCTIONS_SELECTION_HUB_TAB &&
       (isEmpty(hubFunctions) || isEmpty(hubFunctionsCatalog))
     ) {
-      dispatch(functionsActions.fetchHubFunctions()).then(templatesObject => {
+      dispatch(
+        functionsActions.fetchHubFunctions(
+          null,
```
Comment (illia-prokopchuk) [inline src/components/JobWizard/JobWizardSteps/JobWizardFunctionSelection/JobWizardFunctionSelection.js]: change to {}

File: src/components/FeatureStore/Features/Features.js [hunk 3]
```diff
@@ -130,15 +132,17 @@ const Features = ({
               features,
               setFeatures,
               abortControllerRef,
-              setLargeRequestErrorMessage
+              setRequestErrorMessage,
+              result?.filter((resp) => resp.reason)?.[0]?.reason || null,
```
Comment (illia-prokopchuk) [inline src/components/FeatureStore/Features/Features.js]: move to variable

File: src/actions/workflow.js [hunk 4]
```diff
@@ -71,21 +71,37 @@ const workflowActions = {
...
+            errorHandler(response.error)
+          } else {
+            result = result.concat(response.data.runs)
+            nextPageToken = response.data.next_page_token
+          }
+        } catch (error) {
+          nextPageToken = null
+          errorHandler(error)
         }
+
```
Comment (illia-prokopchuk) [inline src/actions/workflow.js]: remove empty line

File: src/actions/workflow.js [hunk 5]
```diff
@@ -71,21 +71,37 @@ const workflowActions = {
...
+    config?.ui?.setRequestErrorMessage?.('')
+
+    const errorHandler = (error) => {
+      const errorMessage = 'Failed to fetch workflows'
+          
+          dispatch(workflowActions.fetchWorkflowsFailure(error))
+          largeResponseCatchHandler(error, errorMessage, dispatch, () => {
+            config?.ui?.setRequestErrorMessage?.(errorMessage)
+          })
+    }
```
Comment (illia-prokopchuk) [inline src/actions/workflow.js:83]: reformat

File: src/utils/largeResponseCatchHandler.js [hunk 6]
```diff
@@ -20,12 +20,14 @@ such restriction.
...
+export const largeResponseCatchHandler = (error, defaultError, dispatch, handlerCallback = () => {}) => {
   if (
     ![LARGE_REQUEST_CANCELED, REQUEST_CANCELED, DEFAULT_ABORT_MSG].includes(error.message) &&
     error &&
     dispatch
   ) {
     showErrorNotification(dispatch, error, defaultError)
+    handlerCallback()
   }
+
```
Comment (illia-prokopchuk) [inline src/utils/largeResponseCatchHandler.js]: remove empty line

File: src/components/ProjectsPage/Projects.js [hunk 7]
```diff
@@ -413,16 +419,17 @@ such as jobs, artifacts, and features.`,
...
       filteredProjects={filteredProjects}
+      filterMatches={filterMatches}
       handleCreateProject={handleCreateProject}
       handleSearchOnFocus={handleSearchOnFocus}
       handleSelectSortOption={handleSelectSortOption}
       isDescendingOrder={isDescendingOrder}
+      projectsRequestErrorMessage={projectsRequestErrorMessage}
       projectStore={projectStore}
       refreshProjects={refreshProjects}
       removeNewProjectError={removeNewProjectError}
```
Comment (illia-prokopchuk) [inline src/components/ProjectsPage/Projects.js:435]: sort props

Comment (Taras-Hlukhovetskyi) [reply src/components/ProjectsPage/Projects.js:435]: alrady sorted

**Pull request closed without merging**